### ISSUE TITLE
Request: .world more dynamic

### ISSUE THREAD

**Issue by willianholtz** [mapeditor/tiled#3338]
Currently when we start a world, we must create a new map, and save it before adding it to the world. To make this more dynamic and improve productivity, it would be interesting to do two things:

* When working on a .world, it would be interesting to be able to add (by right-clicking) a new map without having to save it first, as LDtk does.

* As we are working with .World, it would be interesting to use the programmed layers for the world, and not individually as it is currently. With this, all maps created in .world would have the same layers without the need for the user to spend time recreating everything.

Thus, when saving the .world project, everything created in it is saved in the already configured files (.json, tmj or tmx)

Just like adding the grid to the .world #3258

**Using ideas from other tools is not wanting to be the same, it's improving what's already great!**

**Comment by bjorn** [comment]
> * When working on a .world, it would be interesting to be able to add (by right-clicking) a new map without having to save it first,

I agree. I wrote about possible improvements to the world feature on [this wiki page](https://github.com/mapeditor/tiled/wiki/Enhanced-World-Editing).

> * As we are working with .World, it would be interesting to use the programmed layers for the world, and not individually as it is currently.

Each map having its own set of layers is a flexibility we inherited from the past, which also has its benefits. Yet, I agree it would be convenient if there was a built-in way to define a set of layers that are automatically added to each new map.

Currently, this can be done using scripting, by connecting a JS function to the [`tiled.assetCreated`](https://www.mapeditor.org/docs/scripting/modules/tiled.html#assetCreated) signal and using [`TileMap.addLayer`](https://www.mapeditor.org/docs/scripting/classes/TileMap.html#addLayer). Alternatively, one could save an empty map with certain layer setup as template to create new maps from.

I think the project file would be a good place to define a set of default layers, or we could alternatively introduce explicit support for "map templates".
